addons/l10n_es_edi_facturae/models/account_move.py

```diff
@@ -379,3 +382,247 @@ def _l10n_es_edi_facturae_render_facturae(self):
+        partner = self._import_get_partner(tree, is_bill)
+        self._import_invoice_facturae_invoices(invoice, partner, tree)
+
+    def _import_get_partner(self, tree, is_bill):
+        # If we're dealing with a vendor bill, then the partner is the seller party, if an invoice then it's the buyer.
+        party = tree.xpath('//SellerParty') if is_bill else tree.xpath('//BuyerParty')
+        if party:
+            partner_vals = self._import_extract_partner_values(party[0])
+            return self._import_create_or_retrieve_partner(partner_vals)
+        return None
+
+    def _import_extract_partner_values(self, party_node):
+        name = find_xml_value('.//CorporateName|.//Name', party_node)
+        first_surname = find_xml_value('.//FirstSurname', party_node)
+        second_surname = find_xml_value('.//SecondSurname', party_node)
+        phone = find_xml_value('.//Telephone', party_node)
+        mail = find_xml_value('.//ElectronicMail', party_node)
+        country_code = find_xml_value('.//CountryCode', party_node)
+        vat = find_xml_value('.//TaxIdentificationNumber', party_node)
+
+        full_name = ' '.join(part for part in [name, first_surname, second_surname] if part)
+
+        return {'name': full_name, 'vat': vat, 'phone': phone, 'email': mail, 'country_code': country_code}
+
+    def _import_create_or_retrieve_partner(self, partner_vals):
+        name = partner_vals['name']
+        vat = partner_vals['vat']
+        phone = partner_vals['phone']
+        mail = partner_vals['email']
+        country_code = partner_vals['country_code']
+
+        partner = self.env['res.partner']._retrieve_partner(name=name, vat=vat, phone=phone, mail=mail)
+
+        if not partner and name:
+            partner_vals = {'name': name, 'email': mail, 'phone': phone}
+            country = self.env.ref(f'base.{country_code.lower()}', raise_if_not_found=False) if country_code else False
+            if country:
+                partner_vals['country_id'] = country.id
+            partner = self.env['res.partner'].create(partner_vals)
+            if vat and self.env['res.partner']._run_vat_test(vat, country):
+                partner.vat = vat
+
+        return partner
+
+    def _import_invoice_facturae_invoices(self, invoice, partner, tree):
+        invoices = tree.xpath('//Invoice')
+        if not invoices:
+            return
+
+        self._import_invoice_facturae_invoice(invoice, partner, invoices[0])
+
+        # There might be other invoices inside the facturae.
+        for node in invoices[1:]:
+            other_invoice = invoice.create({
+                'journal_id': invoice.journal_id.id,
+                'move_type': invoice.move_type
+            })
+            with other_invoice._get_edi_creation():
+                self._import_invoice_facturae_invoice(other_invoice, partner, node)
+                other_invoice.message_post(body=escape(_("Created from attachment in %s", invoice._get_html_link())))
```

Comment:
I don't quite understand the `escape` here.
Didn't test the UI, only in a console:

```python
In [1]: from markupsafe import escape, Markup

In [2]: from odoo import _

In [3]: escape(_("Created from attachment in %s", Markup("<a href='huhu'>Name</a>")))
Out[3]: Markup('Created from attachment in &lt;a href=&#39;huhu&#39;&gt;Name&lt;/a&gt;')
```

I would expect more something like
```suggestion
                other_invoice.message_post(body=escape(_("Created from attachment in %s")) % invoice._get_html_link())
```

```python
In [4]: escape(_("Created from attachment in %s")) % Markup("<a href='huhu'>Name</a>")
Out[4]: Markup("Created from attachment in <a href='huhu'>Name</a>")
```